File: CPP/coop.cpp
#include <iostream>
using namespace std ;

#include "coop.h"

void Coop::render() {
	if (!this->getCreation() ){
		if (getGrass() ){
			cout << "#" ;
		}else{
			cout << "=" ;
		}
	}
	else
	{
		this->getCreation()->render() ;	
	}
	
}


int Coop::getIdentity() {
	return 1 ;
}

Coop::Coop() {
	this->setGrass(false) ;
}

Coop::Coop(bool _r) {
	this->setGrass(true) ;
}

File: main/land/Coop.java
package main.land;
//#include "coop.h"
//#include "land.h"

public class Coop extends Land{

public void render() {
	if (this.getCreation() == null){
		if (getGrass() ){
			System.out.print("#") ;
		}else{
			System.out.print("=") ;
		}
	}
	else
	{
		this.getCreation().render() ;	
	}
	
}

public int getIdentity() {
	return 1 ;
}

public Coop() {
	this.setGrass(false) ;
}

public Coop(boolean _r) {
	this.setGrass(true) ;
}

}
